cmd/soroban-rpc/internal/jsonrpc.go

```diff
@@ -42,6 +44,44 @@ type HandlerParams struct {
 	Daemon            interfaces.Daemon
 }
 
+type handlerWithMetrics struct {
+	handler       jrpc2.Handler
+	endpoint      string
+	requestMetric *prometheus.SummaryVec
+}
+
+func (h *handlerWithMetrics) Handle(ctx context.Context, r *jrpc2.Request) (interface{}, error) {
+	startTime := time.Now()
+	result, err := h.handler.Handle(ctx, r)
+	duration := time.Since(startTime).Seconds()
+	label := prometheus.Labels{"status": "ok"}
+	simulateTransactionResponse, ok := result.(methods.SimulateTransactionResponse)
+	if err != nil || ok && simulateTransactionResponse.Error != "" {
+		label = prometheus.Labels{"status": "error"}
+	}
```

Comment:
Shouldn't the same be done for the generic JSON-RPC `Response.Error()` field?

https://pkg.go.dev/github.com/creachadair/jrpc2#Response.Error

Reply:
We may actually want to record the error codes separately